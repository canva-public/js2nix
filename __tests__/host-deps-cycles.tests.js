// @ts-check

const { hostDependencyCycles } = require('../lib/print');
const { renderSnapshot } = require('./fixtures/util');
const abc = require('./fixtures/abc');

describe('Host dependency cycle', () => {
  const cases = [
    'none', //
    'direct', //
    'transitive', //
    'embedded', //
    'overlap', //
  ].map(n => [n, abc[n]]);

  it.each(cases)('the %p case', (_, generator) => {
    const tree = hostDependencyCycles(generator());
    expect(renderSnapshot(tree)).toMatchSnapshot();
  });
});
