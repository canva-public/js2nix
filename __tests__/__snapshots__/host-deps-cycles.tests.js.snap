// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host dependency cycle the "direct" case 1`] = `
"A: 
- A+B

B: 
- A

A+B: 
- A+A

A+A:
"
`;

exports[`Host dependency cycle the "embedded" case 1`] = `
"A: 
- B

B: 
- B+C

C: 
- C+D

D: 
- E

E: 
- C
- F

F: 
- B

C+D: 
- C+E

C+E: 
- F
- C+C

C+C:

B+C: 
- B+D

B+D: 
- B+E

B+E: 
- B+F
- C+C

B+F: 
- B+B

B+B:
"
`;

exports[`Host dependency cycle the "none" case 1`] = `
"A: 
- B

B: 
- C

C: 
- D

D:
"
`;

exports[`Host dependency cycle the "overlap" case 1`] = `
"A: 
- B

B: 
- C

C: 
- C+D

D: 
- E

E: 
- C
- E+F

F: 
- E

C+D: 
- C+E

C+E: 
- C+F
- C+C

C+C:

C+F: 
- C+E

E+F: 
- E+E

E+E:
"
`;

exports[`Host dependency cycle the "transitive" case 1`] = `
"A: 
- B

B: 
- B+C

C: 
- D

D: 
- B

B+C: 
- B+D

B+D: 
- B+B

B+B:
"
`;
